lib/ansible/modules/apt_repository.py

```diff
@@ -688,15 +708,17 @@ def main():
     sources_after = sourceslist.dump()
     changed = sources_before != sources_after
 
+    diff = []
+    sources_added = set()
+    sources_removed = set()
     if changed and module._diff:
-        diff = []
-        for filename in set(sources_before.keys()).union(sources_after.keys()):
+        sources_added = set(sources_after.keys()).difference(sources_before.keys())
```

Comment:
meant the iff for the file diffs, so we always return 'the file names' (which is small amount of data) but file diffs can be a lot and should only return when requested